app/src/main/java/org/hackillinois/androidapp2019/viewmodel/ScannerViewModel.kt

```diff
@@ -20,17 +20,29 @@ import retrofit2.Response
 class ScannerViewModel : ViewModel() {
     private val eventRepository = EventRepository.instance
     var eventsListLiveData: LiveData<List<Event>> = MutableLiveData<List<Event>>()
-    var lastScanStatus: MutableLiveData<ScanStatus> = MutableLiveData<ScanStatus>()
+    var lastScanStatus: MutableLiveData<ScanStatus> = MutableLiveData()
     var shouldDisplayOverrideSwitch = MutableLiveData<Boolean>()
 
-    val CHECK_IN_TEXT = "Check In"
+    private val CHECK_IN_TEXT = "Check In"
 
     fun init() {
         eventsListLiveData = eventRepository.fetchAllEvents()
         // Hidden by default
         shouldDisplayOverrideSwitch.postValue(false)
     }
 
+    fun checkUserIntoEvent(eventName: String, userId: String, staffOverride: Boolean) {
+        if (eventName == CHECK_IN_TEXT) {
+            val hasCheckedIn = true
+            val hasPickedUpSwag = true
+            val checkIn = CheckIn(userId, staffOverride, hasCheckedIn, hasPickedUpSwag)
```

Comment:
Doesn't `hasCheckedIn` imply `hasPickedUpSwag`? Do we need separate variables?

Reply:
We actually never use these two flags - perhaps we want to remove them.
They are not a part of UI anywhere, so perhaps we should discard these from the API entirely.

Reply:
@ASankaran Can you confirm ^?

Reply:
We didn't use them last year. It's up to @bstrauch24 how you handle it this year. You could make it config file based like registration and RSVP for flexibility.